test/cppsuite/tests/search_near_03.cxx

```diff
@@ -0,0 +1,297 @@
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/workload/random_generator.h"
+
+using namespace test_harness;
+
+/*
+ * In this test, we want to verify search_near with prefix enabled when performing unique index
+ * insertions. During the test duration:
+ *  - N threads will keep performing unique index insertions on existing keys in the table, in which
+ * we are expecting it to fail.
+ *  - M threads traverse all the keys in all the collections, to make sure that the number of
+ * collections do not change.
+ */
+class search_near_03 : public test_harness::test {
+    /* a 2D array consisted of a mapping between collection and their inserted prefixes. */
+    std::vector<std::vector<std::string>> prefixes_map;
+    const std::string ALPHABET{"abcdefghijklmnopqrstuvwxyz"};
+
+    public:
+    search_near_03(const test_harness::test_args &args) : test(args) {}
+
+    void
+    run() override final
+    {
+        test::run();
+    }
+
+    /*
+     * Here's how we insert a unique index:
+     * 1. Insert the prefix.
+     * 2. Remove the prefix.
+     * 3. Search near for the prefix. In the case we find a record, we stop here as a value with the
+     * prefix already exists in the table. Otherwise if the record is not found, we can proceed to
+     * insert the full value.
+     * 4. Insert the full value (prefix, id).
+     * All of these operations are wrapped in the same transaction.
+     */
+    static bool
+    perform_unique_index_insertions(
+      thread_context *tc, scoped_cursor &cursor, collection &coll, std::string &prefix_key)
+    {
+        const char *key_tmp;
+        int exact_prefix, ret;
+
+        /* Insert the prefix. */
+        if (!tc->insert(cursor, coll.id, prefix_key))
+            return false;
+
+        /* Remove the prefix. */
+        if (!tc->remove(cursor, coll.id, prefix_key))
+            return false;
+
+        /*
+         * Prefix search near the prefix. We expect that the key is deleted and a WT_NOTFOUND error
+         * to be returned. If the key is present, it means the (prefix, id) has been inserted
+         * already, double check that the prefix portion match.
+         */
+        cursor->reconfigure(cursor.get(), "prefix_search=true");
+        cursor->set_key(cursor.get(), prefix_key.c_str());
+        ret = cursor->search_near(cursor.get(), &exact_prefix);
+        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+        if (ret == 0) {
+            cursor->get_key(cursor.get(), &key_tmp);
```

Comment:
Can we assert here that the key size returned is a full key. If it returned a prefix here, e.g. `AAAA` that would be a different but very significant bug which I think we'd ignore by accident currently.

Reply:
I have modified the `get_prefix_from_key()` to return by nullptr if the format of the key doesn't match. Let me know if thats okay for testing this.

Reply:
Looks fine to me.